core/src/main/java/io/aiven/klaw/service/KafkaConnectControllerService.java

```diff
@@ -911,8 +911,9 @@ private ApiResponse processRestErrorResponse(HttpStatusCodeException e, String d
   }
 
   private boolean checkInPromotionOrder(String envId, String orderOfEnvs) {
-    List<String> orderedEnv = Arrays.asList(orderOfEnvs.split(","));
-    return orderedEnv.contains(envId);
+    return orderOfEnvs.startsWith(envId + ",")
```

Comment:
This looks good but am just testing one scenario where there is only 1 connector env specified, I'm not sure if it will be caught by line 914 it might not match.

Reply:
Yeah I think you need to add an || orderOfEnvs.equals(envId)

as when just one is specified it doesnt have the comma.

Reply:
yes, good catch, thank you